node/node.go

```diff
@@ -608,6 +609,13 @@ type NodeAPI struct { //nolint:golint
 }
 
 func handleNoRoute(c *gin.Context) {
+	matched, _ := regexp.MatchString(`^/v[0-9]+/`, c.Request.URL.Path)
```

Comment:
The change looks good to me.  Nevertheless I think it's worth considering moving this logic of defining the NoRoute handler to the api module, and call the `engine.NoRoute` inside `api.NewAPI`.